File: springboot/src/main/java/com/edufelip/meer/domain/auth/LoginUseCase.java
package com.edufelip.meer.domain.auth;

import com.edufelip.meer.core.auth.AuthUser;
import com.edufelip.meer.domain.repo.AuthUserRepository;
import com.edufelip.meer.security.token.TokenProvider;
import org.springframework.security.crypto.password.PasswordEncoder;

public class LoginUseCase {
    private final AuthUserRepository authUserRepository;
    private final PasswordEncoder passwordEncoder;
    private final TokenProvider tokenProvider;

    public LoginUseCase(AuthUserRepository authUserRepository, PasswordEncoder passwordEncoder, TokenProvider tokenProvider) {
        this.authUserRepository = authUserRepository;
        this.passwordEncoder = passwordEncoder;
        this.tokenProvider = tokenProvider;
    }

    public AuthResult execute(String email, String password) {
        AuthUser user = authUserRepository.findByEmail(email);
        if (user == null) throw new InvalidCredentialsException();
        if (!passwordEncoder.matches(password, user.getPasswordHash())) throw new InvalidCredentialsException();

        String access = tokenProvider.generateAccessToken(user);
        String refresh = tokenProvider.generateRefreshToken(user);

        AuthenticatedUser authUser = new AuthenticatedUser(user.getId(), user.getDisplayName(), user.getEmail());
        return new AuthResult(access, refresh, authUser);
    }
}

File: springboot/src/main/kotlin/com/edufelip/meer/domain/auth/LoginUseCase.kt
package com.edufelip.meer.domain.auth

import com.edufelip.meer.domain.AuthUserRepository
import com.edufelip.meer.security.token.TokenProvider
import org.springframework.security.crypto.password.PasswordEncoder

data class AuthenticatedUser(
    val id: Int,
    val name: String?,
    val email: String?
)

data class AuthResult(
    val token: String,
    val refreshToken: String?,
    val user: AuthenticatedUser
)

class LoginUseCase(
    private val authUserRepository: AuthUserRepository,
    private val passwordEncoder: PasswordEncoder,
    private val tokenProvider: TokenProvider
) {
    fun execute(email: String, password: String): AuthResult {
        val user = authUserRepository.findByEmail(email) ?: throw InvalidCredentialsException()
        if (!passwordEncoder.matches(password, user.passwordHash)) throw InvalidCredentialsException()

        val accessToken = tokenProvider.generateAccessToken(user)
        val refreshToken = tokenProvider.generateRefreshToken(user)

        return AuthResult(
            token = accessToken,
            refreshToken = refreshToken,
            user = AuthenticatedUser(id = user.id, name = user.displayName, email = user.email)
        )
    }
}

class InvalidCredentialsException : RuntimeException("Invalid credentials")
